src/Button/index.tsx

```diff
@@ -1,26 +1,63 @@
 import React from 'react'
 import styled from 'styled-components'
 
-export interface ButtonTextProps {
+type buttonVariant = 'rounded' | 'square' | 'roundedShadow' | 'squareShadow'
+
+export type ButtonTextProps = {
   styles?: React.CSSProperties
   type?: 'submit'
   name?: string
+  variant?: buttonVariant
   onClick?: (e: any) => void
 }
 
 const ButtonBasic: React.FC<React.PropsWithChildren<ButtonTextProps>> = ({
   children,
+  variant,
   ...props
-}) => <Button {...props}>{children}</Button>
+}) => {
+  let Button: React.FunctionComponent<
+    React.ButtonHTMLAttributes<HTMLButtonElement>
+  > | null = null
+
+  switch (variant) {
+    case 'rounded':
+      Button = RoundedButton
+      break
+    case 'roundedShadow':
+      Button = RoundedShadowButton
+      break
+    case 'square':
+      Button = SquareButton
+      break
+    case 'squareShadow':
+      Button = SquareShadowButton
+      break
+    default:
+      Button = RoundedShadowButton
+      break
+  }
+
+  return <Button {...props}>{children}</Button>
+}
 
-export const Button = styled('button')`
+const SquareButton = styled.button`
```

Comment:
Is `styled('button')` same as `styled.button`?

Reply:
yep